spec/models/export/activity_forecast_columns_spec.rb

```diff
@@ -3,60 +3,195 @@
+      end
+
+      it "contains the financial data for financial quarter 4 2021-2022" do
+        expect(value_for_header("Forecast FQ4 2021-2022").to_s).to eql("20000.0")
+      end
+
+      it "contains a zero for the financial quarters inbetween in which there are no forecasts" do
+        expect(value_for_header("Forecast FQ2 2021-2022").to_s).to eql "0"
+        expect(value_for_header("Forecast FQ3 2021-2022").to_s).to eql "0"
+      end
+
+      it "includes a row for each acitvity" do
+        expect(subject.rows.count).to eq(5)
+      end
     end
 
-    it "contains the financial data for financial quarter 4 2021-2022" do
-      expect(value_for_header("Forecast FQ4 2021-2022").to_s).to eql("10000.0")
+    context "when there are no activities" do
+      subject { Export::ActivityForecastColumns.new(activities: []) }
+
+      it "returns an empty array" do
+        expect(subject.headers).to eql []
+        expect(subject.rows).to eql []
+      end
     end
+  end
+
+  context "when a report is passed in" do
+    let(:report) { create(:report, financial_quarter: 1, financial_year: 2019) }
 
-    it "contains a zero for the financial quarters inbetween in which there are no forecasts" do
-      expect(value_for_header("Forecast FQ2 2021-2022").to_s).to eql "0"
-      expect(value_for_header("Forecast FQ3 2021-2022").to_s).to eql "0"
+    describe "#headers" do
+      it "incudes the heading that describe the finances for the finnacial quarter of the report" do
+        expect(subject.headers).to include("Forecast FQ2 2019-2020")
+      end
+
+      it "includes the heading that describe the finances for financial quarter FQ1 2021-2022" do
+        expect(subject.headers).to include(
+          "Forecast FQ1 2021-2022",
+        )
+      end
+
+      it "includes the heading that describe the finances for financial quarter FQ4 2021-2022" do
+        expect(subject.headers).to include(
+          "Forecast FQ4 2021-2022",
+        )
+      end
+
+      it "includes the headings that describe the finances for financial quarters inbetween" do
+        expect(subject.headers).to include(
+          "Forecast FQ2 2021-2022",
+          "Forecast FQ3 2021-2022",
+        )
+      end
+
+      it "does not include the headers for the quarter before the report" do
+        expect(subject.headers).not_to include("Forecast FQ4 2018-2019")
+      end
     end
 
-    it "includes a row for each acitvity" do
-      expect(subject.rows.count).to eq(5)
+    describe "#rows" do
+      it "contains the financial data for financial quarter of the report" do
+        expect(value_for_header("Forecast FQ1 2019-2020").to_s).to eql("0")
+      end
+
+      it "contains the financial data for financial quarter 1 2021-2022" do
+        expect(value_for_header("Forecast FQ1 2021-2022").to_s).to eql("20000.0")
+      end
+
+      it "contains the financial data for financial quarter 4 2021-2022" do
```

Comment:
I wonder if it's worth being really explicit here that the forecast data returned is for the financial period, as reported in the given report as distinct from the forecast given for the same period in other reports?

e.g. something like:

```
"contains the the forecast for Q1 2021-2022 as provided in the given report"
```